relay-server/src/extractors/signed_json.rs

```diff
@@ -23,6 +23,8 @@ pub struct SignedJson<T> {
 enum SignatureError {
     #[fail(display = "invalid relay signature")]
     BadSignature,
+    #[fail(display = "could not deserialize payload")]
```

Comment:
"payload" here is ambiguous to the user, probably we should name this "signature".

Reply:
The differentiation is about getting an error because you have the wrong credentials (you are not authorised) or you are asking for the wrong stuff ( there is no problem with your credentials, you just misspelled a property name in your payload).  
This is exactly what happened to me, I placed in the request the relay_id instead of the dsn and there was an error deserializing (because the size of the string was wrong) and I was getting a 403 bad signature and could not figure out what was wrong with the signing of the request (everything seemed fine, I signed it manually... and verified... yet
I was getting a 403 with a "invalid signature")

Reply:
To the user, the signature is opaque. It's just a bunch of bytes, and the bytes aren't a valid signature. We can surely have more precise error information internally, but I wouldn't expose this difference via the API. That creates a too strict contract.

If you're working on the signature, you're effectively opening the box, so you can use unit tests to get the precise internal signature error.